planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/utils.cpp

```diff
@@ -1294,4 +1096,38 @@ bool object_path_overlaps_lanes(
     return !boost::geometry::disjoint(path, lanes_polygon);
   });
 }
+
+std::vector<std::vector<PoseWithVelocityStamped>> convert_to_predicted_paths(
+  const CommonDataPtr & common_data_ptr, const LaneChangePath & lane_change_path,
+  const size_t deceleration_sampling_num)
+{
+  static constexpr double floating_err_th{1e-3};
+  const auto bpp_param = *common_data_ptr->bpp_param_ptr;
+  const auto global_min_acc = bpp_param.min_acc;
+  const auto lane_changing_acc = lane_change_path.info.longitudinal_acceleration.lane_changing;
+
+  const auto min_acc = std::min(lane_changing_acc, global_min_acc);
+  const auto sampling_num =
+    std::abs(min_acc - lane_changing_acc) > floating_err_th ? deceleration_sampling_num : 1;
+  const auto acc_resolution = (min_acc - lane_changing_acc) / static_cast<double>(sampling_num);
+
+  const auto ego_predicted_path = [&](size_t n) {
+    auto acc = lane_changing_acc + static_cast<double>(n) * acc_resolution;
+    return utils::lane_change::convert_to_predicted_path(common_data_ptr, lane_change_path, acc);
+  };
+
+  return ranges::views::iota(0UL, sampling_num) | ranges::views::transform(ego_predicted_path) |
+         ranges::to<std::vector>();
+}
+
+bool is_valid_start_point(const lane_change::CommonDataPtr & common_data_ptr, const Pose & pose)
+{
+  const lanelet::BasicPoint2d lc_start_point(pose.position.x, pose.position.y);
+
+  const auto & target_neighbor_poly = common_data_ptr->lanes_polygon_ptr->target_neighbor;
+  const auto & target_lane_poly = common_data_ptr->lanes_polygon_ptr->target;
+
+  return boost::geometry::covered_by(lc_start_point, target_neighbor_poly) ||
+         boost::geometry::covered_by(lc_start_point, target_lane_poly);
```

Comment:
I would expect the start point to be valid if it is in the current ego lane, not in the target lane :thinking:

Reply:
For target lane, we consider that the reference path where we get the prepare segment might be shifted by avoidance module.

Reply:
Added comment in https://github.com/autowarefoundation/autoware.universe/pull/9633/commits/264733480ecd8bbb3c4c12d25fcc7b658465381f